tests/test_sailor/test_sap_iot/test_wrappers.py

```diff
@@ -102,6 +102,47 @@ def test_interpolate_missing_data(simple_dataset, method, expect_ffill, expect_b
         assert not interpolated_dataset._df[interpolated_dataset._df.timestamp > actual_end].isnull().any().any()
 
 
+def test_filter_equipment_id_only(simple_dataset):
+    filtered_dataset = simple_dataset.filter(equipment_ids=['equipment_id_1'])
+
+    assert len(filtered_dataset.equipment_set) == 1
+    assert len(filtered_dataset._df) == 100
+    assert filtered_dataset._df.equipment_id.unique() == ['equipment_id_1']
+    assert filtered_dataset.indicator_set == simple_dataset.indicator_set
+
+
+def test_filter_equipment_set_only(simple_dataset):
+    filtered_dataset = simple_dataset.filter(equipment_set=simple_dataset.equipment_set.filter(id='equipment_id_1'))
+
+    assert len(filtered_dataset.equipment_set) == 1
+    assert len(filtered_dataset._df) == 100
+    assert filtered_dataset._df.equipment_id.unique() == ['equipment_id_1']
+    assert filtered_dataset.indicator_set == simple_dataset.indicator_set
+
+
+def test_filter_equipment_set_and_id(simple_dataset):
+    filtered_dataset = simple_dataset.filter(equipment_ids=['equipment_id_2'],
+                                             equipment_set=simple_dataset.equipment_set.filter(id='equipment_id_1'))
+
+    assert len(filtered_dataset.equipment_set) == 1
+    assert len(filtered_dataset._df) == 100
+    assert filtered_dataset._df.equipment_id.unique() == ['equipment_id_1']
+    assert filtered_dataset.indicator_set == simple_dataset.indicator_set
+
+
+def test_filter_indicator_set(simple_dataset):
+    selected_indicators = simple_dataset.indicator_set.filter(id='indicator_id_1')
+
+    filtered_dataset = simple_dataset.filter(indicator_set=selected_indicators)
+
+    assert len(filtered_dataset.equipment_set) == 2
+    assert len(filtered_dataset._df) == len(simple_dataset._df)
+    assert filtered_dataset.indicator_set == selected_indicators
+    assert len(filtered_dataset._df.columns) == 4
+    assert all(filtered_dataset._df.columns == (filtered_dataset.get_index_columns() +
+                                                [indicator._unique_id for indicator in selected_indicators]))
+
+
```

Comment:
I just put that comment here because it is related to testing:

The tests below emit the new warning. Make sure to ignore it where you explicitly use the filtering on ids. I guess the test_interpolate_missing_data should rather be converted to use the filtering using sets.

```
tests/test_sailor/test_sap_iot/test_wrappers.py::test_interpolate_missing_data[pad-True-False]
tests/test_sailor/test_sap_iot/test_wrappers.py::test_interpolate_missing_data[slinear-False-False]
tests/test_sailor/test_sap_iot/test_wrappers.py::test_interpolate_missing_data[bfill-False-True]
tests/test_sailor/test_sap_iot/test_wrappers.py::test_filter_equipment_id_only
tests/test_sailor/test_sap_iot/test_wrappers.py::test_filter_equipment_set_and_id
```

Reply:
thanks, forgot to check for the warnings -- sorry :)